widgetssdk/src/main/java/com/glia/widgets/launcher/EngagementLauncher.kt

```diff
@@ -2,50 +2,96 @@ package com.glia.widgets.launcher
 
 import android.content.Context
 import com.glia.androidsdk.Engagement
+import com.glia.widgets.callvisualizer.controller.CallVisualizerContract
 import com.glia.widgets.chat.Intention
 import com.glia.widgets.core.secureconversations.domain.HasOngoingSecureConversationUseCase
-import com.glia.widgets.di.ControllerFactory
 import com.glia.widgets.engagement.domain.EndEngagementUseCase
+import com.glia.widgets.engagement.domain.EngagementTypeUseCase
 import com.glia.widgets.engagement.domain.IsQueueingOrLiveEngagementUseCase
 import com.glia.widgets.helper.Logger
 import com.glia.widgets.helper.TAG
 
+@RequiresOptIn(
+    level = RequiresOptIn.Level.ERROR,
+    message = "This feature is under consideration for removal in a future releases. Use with caution."
+)
+@Retention(AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY)
+/**
+ * Marks the feature as under consideration for removal in a future releases.
+ */
+annotation class FeatureUnderConsiderationForRemoval
+
 /**
  * An interface for launching different types of engagements, such as chat,
  * audio calls, video calls, and secure messaging
  */
+@OptIn(FeatureUnderConsiderationForRemoval::class)
 interface EngagementLauncher {
+
+    /**
+     * Starts a chat engagement.
+     *
+     * @param context Activity or Context used to launch the chat screen
+     */
+    fun startChat(context: Context) = startChat(context, null)
+
     /**
      * Starts a chat engagement.
      *
      * @param context Activity or Context used to launch the chat screen
      * @param visitorContextAssetId Optional visitor context id from Glia Hub
      */
-    fun startChat(context: Context, visitorContextAssetId: String? = null)
+    @FeatureUnderConsiderationForRemoval
```

Comment:
What do you think about this solution?
As the `@JvmOverloads` annotation is not supported for interfaces, integrators using Java must explicitly pass null for `visitorContextAssetId`. For this reason, I have defined a new function instead.
Later I added this annotation to prevent integrators from using the functions with `visitorContextAssetId` until they need it. To do that, they have to mark their functions with the `@OptIn(FeatureUnderConsiderationForRemoval::class)` annotation, otherwise it won't compile.